### PR TITLE
refactor: migrate UI to material3 [AR-3442]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-3442" title="AR-3442" target="_blank">TaskAR-3442</a>  Migrate to material3
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We are still mixing material2 and material3 components, it appeared recently on the bottom sheet in dark mode with texts being dark on dark background and it's because we use material2 bottom sheet but in its content we use material3 texts.

### Solutions

We should migrate to material3 only, now it should be possible as I checked and almost all material2 components that we still use have already the equivalent in material3 (only `swipeable` is still missing, or rather marked as internal as they are still making it stable enough).

Sorry about the size of it, but mainly it's changing the imports and "experimental" annotations. The elements that have actually been modified are the bottom sheets as they work in a different way in material3.

#### How to Test

Open the app in light and dark theme and check if it looks like on the designs.

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt [hunk 1]
```diff
@@ -135,14 +133,14 @@ private fun OngoingCallContent(
...
         initialValue = sheetInitialValue
     ).also {
-        LaunchedEffect(Unit) {
-            // Same issue with expanded on other sheets, we need to use animateTo to fully expand programmatically.
-            it.animateTo(sheetInitialValue)
-        }
+//        LaunchedEffect(Unit) {
+//            // Same issue with expanded on other sheets, we need to use animateTo to fully expand programmatically.
+//            it.show()
+//        } TODO
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt]: Not needed ? maybe we remove it if so

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1811/1/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallViewModel.kt [hunk 2]
```diff
@@ -48,7 +47,7 @@ import javax.inject.Inject
 
 @Suppress("LongParameterList")
 @HiltViewModel
-class OngoingCallViewModel @OptIn(ExperimentalCoroutinesApi::class)
+class OngoingCallViewModel
 @Inject constructor(
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallViewModel.kt]: can you just make it one line

File: app/src/main/kotlin/com/wire/android/ui/home/HomeStateHolder.kt [hunk 3]
```diff
@@ -78,13 +75,13 @@ class HomeStateHolder(
...
         coroutineScope.launch {
-            if (!bottomSheetState.isVisible) bottomSheetState.animateTo(ModalBottomSheetValue.Expanded)
+            if (!bottomSheetState.isVisible) bottomSheetState.show()
         }
     }
 
     fun closeBottomSheet() {
         coroutineScope.launch {
-            if (bottomSheetState.isVisible) bottomSheetState.animateTo(ModalBottomSheetValue.Hidden)
+            if (bottomSheetState.isVisible) bottomSheetState.hide()
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/home/HomeStateHolder.kt:84]: No animation ?

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreenState.kt [hunk 4]
```diff
@@ -63,40 +62,39 @@ fun rememberConversationScreenState(
...
+            modalBottomSheetState.hide()
             onComplete()
         }
     }
 
     fun copyMessage(text: String) {
         clipboardManager.setText(AnnotatedString(text))
         coroutineScope.launch {
-            modalBottomSheetState.animateTo(ModalBottomSheetValue.Hidden)
+            modalBottomSheetState.hide()
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreenState.kt:90]: You are removing animation here as well

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreenState.kt [hunk 5]
```diff
@@ -63,40 +62,39 @@ fun rememberConversationScreenState(
...
             snackBarHostState.showSnackbar(context.getString(R.string.info_message_copied))
         }
     }
 
     fun showSelfDeletionContextMenu() {
         bottomSheetMenuType = BottomSheetMenuType.SelfDeletion
-        coroutineScope.launch { modalBottomSheetState.animateTo(ModalBottomSheetValue.Expanded) }
+        coroutineScope.launch { modalBottomSheetState.show() }
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreenState.kt:97]: and here

**Review by ohassine — APPROVED** [review]
Well done 🧹 just left some tiny comments

File: app/src/main/kotlin/com/wire/android/ui/home/HomeStateHolder.kt (hunk 3 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/home/HomeStateHolder.kt:84]: `animateTo` is `internal` in material3 but we have `hide`, `show`, `expand` and `partialExpand` functions, all of which execute `animateTo` under the hood 😃

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreenState.kt (hunk 4 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreenState.kt:90]: `animateTo` is `internal` in material3 but we have `hide`, `show`, `expand` and `partialExpand` functions, all of which execute `animateTo` under the hood 😃

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreenState.kt (hunk 5 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreenState.kt:97]: `animateTo` is `internal` in material3 but we have `hide`, `show`, `expand` and `partialExpand` functions, all of which execute `animateTo` under the hood 😃

**Review by trOnk12 — APPROVED** [review]
great effort <3 kudos to you

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt]: Oh, I forgot, that's why I'm adding these "TODO"s to code but this time when committing I didn't get the "TODO" warning, don't know why, so it slipped through 🤔 I commented that at the beginning as I wasn't sure how the "show" should be implemented in material3, but I fixed that already.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1811/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6588234)(6588234)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6588234)(6588234)-staging-compat.apk)

**Pull request merged**